internal/server/singleprocess/service_status_report.go

```diff
@@ -77,3 +78,75 @@ func (s *service) GetStatusReport(
 
 	return r, nil
 }
+
+// Builds a status report job, queues it, and returns the job ID
+func (s *service) ExpediteStatusReport(
+	ctx context.Context,
+	req *pb.ExpediteStatusReportRequest,
+) (*pb.ExpediteStatusReportResponse, error) {
+	statusReportJob := &pb.Job_StatusReport{
+		StatusReport: &pb.Job_StatusReportOp{},
+	}
+
+	var applicationRef *pb.Ref_Application
+	switch target := req.Target.(type) {
+	case *pb.ExpediteStatusReportRequest_Deployment:
+		applicationRef = target.Deployment.Application
+		statusReportJob.StatusReport.Target = &pb.Job_StatusReportOp_Deployment{
+			Deployment: target.Deployment,
+		}
+	case *pb.ExpediteStatusReportRequest_Release:
+		applicationRef = target.Release.Application
+		statusReportJob.StatusReport.Target = &pb.Job_StatusReportOp_Release{
+			Release: target.Release,
+		}
+	default:
+		return nil, status.Errorf(codes.FailedPrecondition, "unknown status report target: %T", req.Target)
+	}
+
+	// Status report jobs need the parent project to obtain its datasource
+	project, err := s.state.ProjectGet(&pb.Ref_Project{Project: applicationRef.Project})
+	if err != nil {
+		return nil, err
+	}
+
+	// Get target from request
+
+	// build job
+	jobRequest := &pb.QueueJobRequest{
+		Job: &pb.Job{
+			// SingletonId so that we only have one poll operation at
+			// any time queued per application.
+			SingletonId: fmt.Sprintf("status-report-ondemand/%s", applicationRef.Application),
+
+			Application: applicationRef,
+
+			// Application polling requires a data source to be configured for the project
+			// Otherwise a status report can't properly eval the projects hcl context
+			// needed to query the deploy or release
+			DataSource: project.DataSource,
+
+			Workspace: &pb.Ref_Workspace{Workspace: "default"},
```

Comment:
Note: if the UI has the workspace available we should hook up `workspace` as a parameter to `ExpediteStatusReportRequest`

Reply:
The UI technically does have the workspace available (it’s the first chunk of the path i.e. `/default/projects/...`) but it’s not really wired in so this needn’t be a blocker.

Reply:
Ok cool! I think it might be more reliable right now for the UI to request the workspace. I think the only way the server can know is on the Project proto, which is an array of Workspaces. So if the UI already knows what workspace the project is in, that might be the best approach at the moment.

If that makes I'll update the PR to expect a workspace param